### PR TITLE
SPM memory quick fixes and m1 tests

### PR DESCRIPTION
**Pull request by ttoomey**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch

## Description
* Adjust SPM, attempt to clear memoized vars as a low-impact memory fix (low confidence this works)
* Optimizations to episode calculation to reduce memory usage
* Enhances/streamlines SPM m1 tests
* Add test to validate edge cases in SPM
* Add check for cancellation inside HDX upload step
* Support bypass skipping during testkit table comparisons.
* removed low-value tests

### Type of Change
Refactor | Code clean-up | New feature

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] I have updated the documentation (or not applicable)
- [x] I have added spec tests (or not applicable)
- [x] I have provided testing instructions in this PR or the related issue (or not applicable)

### CODE REVIEW

File: drivers/hud_spm_report/spec/models/fy2026/measure_1a_spec.rb [hunk 1]
```diff
@@ -35,14 +35,6 @@
         run_measure(@report, HudSpmReport::Generators::Fy2026::MeasureOne)
       end
 
-      it 'creates SpmEnrollment records correctly' do
```
Comment (ttoomey) [inline drivers/hud_spm_report/spec/models/fy2026/measure_1a_spec.rb:38]: removed low value test

File: drivers/hud_spm_report/app/models/hud_spm_report/fy2026/episode_batch.rb [hunk 2]
```diff
@@ -40,15 +40,19 @@ def calculate_batch(client_ids)
...
+      batch_personal_ids = enrollments_for_clients.values.flat_map { |es| es.map(&:personal_id) }.uniq
       # Load all bed nights for these clients regardless of enrollment; we'll look them up as necessary
       # Bednights are indexed on `[EnrollmentID, PersonalID, data_source_id]`
-      batch_services = GrdaWarehouse::Hud::Service.bed_night.
-        between(start_date: nil, end_date: @filter.end). # We don't need anything after the report end date, but may need services before the start date
+
+      lookback_date = @filter.start - 7.years
+      batch_services = Hash.new { |h, k| h[k] = [] }
+      GrdaWarehouse::Hud::Service.bed_night.
+        between(start_date: lookback_date, end_date: @filter.end). # We don't need anything after the report end date, but may need services before the start date
```
Comment (ttoomey) [inline drivers/hud_spm_report/app/models/hud_spm_report/fy2026/episode_batch.rb:50]: I don't think we need services before lookback date

File: drivers/hud_spm_report/spec/models/fy2026/measure_1b_spec.rb [hunk 3]
```diff
@@ -1251,4 +1307,70 @@
...
+        project: @nbn_project,
+        entry_date: '2021-01-01'.to_date,
+        exit_date: '2023-01-31'.to_date,
+      )
+
+      # Bed nights defining the 365-day window
+      create_bed_night_service(enrollment: enrollment, date: '2022-01-15'.to_date)
+      create_bed_night_service(enrollment: enrollment, date: '2023-01-15'.to_date)
+
+      # Bed night with a 1-day gap (2022-01-14 is missing)
```
Comment (eanders) [inline drivers/hud_spm_report/spec/models/fy2026/measure_1b_spec.rb]: I'm not convinced this test is correct.  I don't believe having a 1 day gap within an ES NbN is sufficient to create an episode.  
To test this, I believe you'd need a 1 day gap between enrollments.

**Review by eanders — APPROVED** [review]
@ttoomey I think this looks reasonable, as we discussed, the contiguity test doesn't quite look right to me, but I'm not seeing the related changes in the code, so maybe it's ok now?

File: drivers/hud_spm_report/spec/models/fy2026/measure_1b_spec.rb (hunk 3 above)
Comment (ttoomey) [reply drivers/hud_spm_report/spec/models/fy2026/measure_1b_spec.rb]: capturing discussion from huddle: we took a second look at these; I think they are real but I'm not going to try and fix them in this PR; will create a follow up
